mesonbuild/interpreter/interpreter.py

```diff
@@ -2850,7 +2856,20 @@ def build_incdir_object(self, incdir_strings: T.List[str], is_system: bool = Fal
             absdir_build = os.path.join(absbase_build, a)
             if not os.path.isdir(absdir_src) and not os.path.isdir(absdir_build):
                 raise InvalidArguments(f'Include dir {a} does not exist.')
-        i = build.IncludeDirs(self.subdir, incdir_strings, is_system)
+
+        if not (build_relative or source_relative):
+            raise InvalidArguments('include_directories must use \'build\'-relative, '
```

Comment:
Not sure about the `-relative`. From the user perspective, there are only `build` and `source` arguments to provide.

Reply:
I see.  I think what's going on is a couple of things that just might not fit entirely perfectly together, at least for you (and maybe others?) -
- I generally lean towards longer names when they can convey a little more meaningful information but I get the impression that others have a low tolerance to them and much prefer shorter names, even when there may sometimes be a little information lost).
- An error message presented to the user seems like a good opportunity to convey as much information and clarity to the user as possible; there's no benefit to being econoimical with error messages.

To me, an IncludeDirs parameter called 'build_relative' is just that little bit more immediately obvious as to what it does than one just called 'build'.  However, 'build' and 'source' had already been suggested and so, rather than risk the ire of any meson gatekeepers from using longer names 😱 , it seemed best to go with what's been suggested, while taking the opportunity to be a bit more verbose/meaningful in the associated error message... but that would appear to have backfired with you 😆

Out of interest, if a variable is called 'build' and you understand it to mean 'build-relative', is it jarring to refer to it as "'build'-relative"?  Also, would you prefer to name it 'build_relative' in all cases or only ever 'build' as its parameter name and in error messages, omitting the opporunity to explicitly clarify that we're really dealing with build-relative paths?

At this point, I really don't mind; whatever helps get this merged.